.github/workflows/lint-server.yml

```diff
@@ -16,7 +19,24 @@ jobs:
         with:
           go-version: 1.18.1
       - uses: actions/checkout@v3
+        with:
+          path: "focalboard"
+      - id: "mattermostServer"
+        uses: actions/checkout@v3
+        continue-on-error: true
+        with:
+          repository: "mattermost/mattermost-server"
+          path: "mattermost-server"
+          ref: ${{ env.BRANCH_NAME }}
+      - uses: actions/checkout@v3
+        if: steps.mattermostServer.outcome == 'failure'
+        with:
+          repository: "mattermost/mattermost-server"
```

Comment:
Same as above: 

```suggestion
          repository: "mattermost/mattermost-server"
          fetch-depth: "20" 
```

Reply:
why not just fetch latest commit?

Reply:
I have seen issues with that approach, with desired commits missing when cloned.
MergeCommits are also counted as an actual commit. 

This is an internal git clone, generated from within gitHub(gitHub runners) towards gitHub. The performance gain should be negligible